lib/index.js

```diff
@@ -214,32 +257,33 @@ Appboy.prototype.identify = function(identify) {
   var phone = identify.phone();
   var traits = clone(identify.traits());
 
-  window.appboy.changeUser(userId);
-  window.appboy.getUser().setAvatarImageUrl(avatar);
-  window.appboy.getUser().setEmail(email);
-  window.appboy.getUser().setFirstName(firstName);
-  window.appboy.getUser().setGender(getGender(gender));
-  window.appboy.getUser().setLastName(lastName);
-  window.appboy.getUser().setPhoneNumber(phone);
-  if (address) {
-    window.appboy.getUser().setCountry(address.country);
-    window.appboy.getUser().setHomeCity(address.city);
-  }
-  if (birthday) {
-    window.appboy.getUser().setDateOfBirth(birthday.getUTCFullYear(), birthday.getUTCMonth() + 1, birthday.getUTCDate());
-  }
-
-  // delete all the standard traits from traits clone so that we can use appboy's setCustomAttribute on non-standard traits
-  // also remove all reserved keys so we dont set them as custom attributes, otherwise Appboy rejects the entire event
-  // https://www.appboy.com/documentation/Platform_Wide/#reserved-keys
-  var reserved = ['avatar', 'address', 'birthday', 'email', 'id', 'firstName', 'gender', 'lastName', 'phone', 'facebook', 'twitter', 'first_name', 'last_name', 'dob', 'external_id', 'country', 'home_city', 'bio', 'gender', 'phone', 'email_subscribe', 'push_subscribe'];
-  each(function(key) {
-    delete traits[key];
-  }, reserved);
-
-  each(function(value, key) {
-    window.appboy.getUser().setCustomUserAttribute(key, value);
-  }, traits);
+  this._changeUser(userId, function() {
```

Comment:
These don't need to go inside of the changeUser messagingReadyCallback callback -- none of these should trigger IAMs.